spec/listing/listing_spec.js

```diff
@@ -1,97 +1,88 @@
-    });
+  describe("inheritance", function() {
+    describe("Item", function() {
+      it("is inheritable", function() {
+        class ChildItem extends Item {
 
-    it("should expose a List", function() {
-      app.view.listing("MyList", {});
-      expect(new app.Views.MyList({ collection: collection })).toEqual(jasmine.any(Listing.List));
+        }
+
+        expect(new ChildItem()).toEqual(jasmine.any(Item));
+      });
     });
-  });
 
-  describe("factories", function() {
+    describe("List", function() {
+      it("is inheritable", function() {
+        class ChildItem extends Item {
 
-    it("should expose #listing", function() {
-      app.view.listing("abc", {});
-      expect(new app.Views.abc({ collection: collection })).toEqual(jasmine.any(baseExports.View));
-    });
+        }
 
-    it("should expose #listing.item", function() {
-      app.view.listing.item("abc", {});
-      expect(new app.Views.abc).toEqual(jasmine.any(baseExports.View));
-    });
+        class ChildList extends List {
+          get itemClass() {
+            return ChildItem;
+          }
+        }
 
-    it("requires that a collection be provided", function() {
-      expect(function() {
-        app.view.listing("abc", {});
-        new app.Views.abc();
-      }).toThrowError("A collection must be provided");
-    });
+        expect(new ChildItem()).toEqual(jasmine.any(Item));
+        expect(new ChildList({collection})).toEqual(jasmine.any(List));
+      });
 
-    it("should get a reference to the item class", function() {
-      // intentionally define the listing before the item is defined to verify lazy evaluation of the item class
-      app.view.listing("abc", {
-        itemClassName: "def"
+      it("requires itemClass to be defined", function() {
+        class ChildList extends List {
+        }
+
+        expect(() => {
+          // eslint-disable-next-line no-new
+          new ChildList({collection});
+        }).toThrowError("itemClass must be defined");
       });
-      app.view.listing.item("def", {});
-      var listing = new app.Views.abc({ collection: collection });
-      expect(listing.itemClass).toEqual(app.Views.def);
     });
-
   });
 
   describe("rendering", function() {
-
-    var listing;
+    let listing;
 
     beforeEach(function() {
-      app.view.listing("Main", {
-        tagName: "ul",
-        itemClassName: "Sub"
-      });
-      app.view.listing.item("Sub", {
-        tagName: "li",
-        render: function() {
-          this.$el.html(this.model.get("name"));
-          return this;
-        }
-      });
-      listing = new app.Views.Main({ collection: collection });
+      listing = new Main({ collection });
       listing.render();
     });
 
     describe("collection is reset", function() {
-
       beforeEach(function() {
-        collection.reset([ { name: "Bob"}, { name: "Joe" } ]);
+        collection.reset([ { name: "Bob" }, { name: "Joe" } ]);
```

Comment:
Our styleguide says no spaces after `[` or before `]` in an array. Not sure why it isn't picked up by ESL.

Reply:
I'll look into updating our EsLint config.